manifests/init.pp

```diff
@@ -40,18 +40,17 @@
   validate_re($cron_allow, '^(present)|(absent)$', "cron::cron_allow is ${cron_allow} and must be absent or present")
   validate_re($cron_deny, '^(present)|(absent)$', "cron::cron_deny is ${cron_deny} and must be absent or present")
 
-  case type($enable_cron) {
-    'string': {
-      validate_re($enable_cron, '^(true|false)$', "cron::enable_cron may be either 'true' or 'false' and is set to <${enable_cron}>")
-      $enable_cron_real = str2bool($enable_cron)
-    }
-    'boolean': {
-      $enable_cron_real = $enable_cron
-    }
-    default: {
-      fail('cron::enable_cron type must be true or false.')
-    }
+  if is_string($enable_cron) {
+    validate_re($enable_cron, '^(true|false)$', "cron::enable_cron may be either 'true' or 'false' and is set to <${enable_cron}>")
```

Comment:
If passing a non-string to str2bool() it will fail. My favorite of stringified boolean treatment only output bools.

Here is a verified example with spec tests for the failing part. It is taken from:
https://github.com/kylefu8/puppet-module-ltscore/blob/master/manifests/init.pp#L21-L27
https://github.com/kylefu8/puppet-module-ltscore/blob/master/spec/classes/init_spec.rb#L436-L453

<pre>
  if is_bool($fix_access_to_alsa) {
    $fix_access_to_alsa_real = $fix_access_to_alsa
  }
  else {
    $fix_access_to_alsa_real = str2bool($fix_access_to_alsa)
  }
</pre>

<pre>
# should fail on invalid types tests
  ['invalid',3,2.42,['array'],a = { 'ha' => 'sh' }].each do |value|
# <fix_access_to_alsa should fail on invalid types>
    context "When fix_access_to_alsa set to invalid #{value} (as #{value.class}) on supported OS" do
      let (:params) { { :fix_access_to_alsa => value } }
      let :facts do
        { :osfamily          => 'Suse',
          :lsbmajdistrelease => '11',
        }
      end

      it 'should fail' do
        expect {
          should
        }.to raise_error(Puppet::Error, /^str2bool\(\):/)
      end
    end
# </fix_access_to_alsa should fail on invalid types>
</pre>